src/main/java/gregtech/api/util/oreglob/OreGlob.java

```diff
@@ -7,37 +7,50 @@
 import org.jetbrains.annotations.ApiStatus;
 import org.jetbrains.annotations.NotNull;
 
+import java.util.Collection;
 import java.util.List;
-import java.util.Set;
-import java.util.function.Function;
 
 /**
  * Glob-like string matcher language designed for ore dictionary matching.
  * <p>
- * An OreGlob instance provides two functions: the ability to match strings,
- * and the ability to translate expression structure into user-friendly text
- * explanations. The text can be either a plaintext, or a text formatted by standard
+ * An OreGlob instance provides two functions: the ability to match strings, and the ability to translate expression
+ * structure into user-friendly text explanations. The text can be either a plaintext, or a text formatted by standard
  * Minecraft text format.
  */
 public abstract class OreGlob {
 
-    private static Function<String, OreGlobCompileResult> compiler;
+    private static OreGlobCompiler compiler;
 
     /**
      * Tries to compile the string expression into OreGlob instance.
      *
      * @param expression OreGlob expression
      * @return Compilation result
      * @throws IllegalStateException If compiler is not provided yet
+     * @deprecated use {@link #compile(String, boolean)}
      */
     @NotNull
+    @Deprecated
```

Comment:
Could the deprecated things here get a scheduled for removal annotation too? (removal in 2.9)